src/layouts/EditPage.jsx

```diff
@@ -128,53 +139,68 @@ const EditPage = ({ match }) => {
         params={decodedParams}
       />
       <div className={elementStyles.wrapper}>
-        {isXSSViolation &&
-          showXSSWarning && ( // to be refactored later
-            <GenericWarningModal
-              displayTitle="Warning"
-              // DOMPurify removed object format taken from https://github.com/cure53/DOMPurify/blob/dd63379e6354f66d4689bb80b30cb43a6d8727c2/src/purify.js
-              displayText={`There is unauthorised JS detected in the following snippet${
-                DOMPurify.removed.length > 1 ? "s" : ""
-              }:
-            ${DOMPurify.removed.map(
-              (elem, i) =>
-                `<br/><code>${i}</code>: <code>${
-                  elem.attribute?.textContent || elem.element?.textContent
-                    ? (
-                        elem.attribute?.textContent || elem.element?.textContent
-                      ).replace("<", "&lt;")
-                    : elem
-                }</code>`
-            )}
-            <br/><br/>Before saving, the editor input will be automatically sanitised to prevent security vulnerabilities.
-            <br/><br/>To save the sanitised editor input, press Acknowledge. To return to the editor without sanitising, press Cancel.`}
-              onProceed={() => {
+        {isXSSViolation && (
+          <GenericWarningModal
+            isOpen={isXSSOpen}
+            onClose={onXSSClose}
+            displayTitle="Warning"
+            // DOMPurify removed object format taken from https://github.com/cure53/DOMPurify/blob/dd63379e6354f66d4689bb80b30cb43a6d8727c2/src/purify.js
+            displayText={`There is unauthorised JS detected in the following snippet${
+              DOMPurify.removed.length > 1 ? "s" : ""
+            }:
+          ${DOMPurify.removed.map(
+            (elem, i) =>
+              `<br/><code>${i}</code>: <code>${
+                elem.attribute?.textContent || elem.element?.textContent
+                  ? (
+                      elem.attribute?.textContent || elem.element?.textContent
+                    ).replace("<", "&lt;")
+                  : elem
+              }</code>`
+          )}
+          <br/><br/>Before saving, the editor input will be automatically sanitised to prevent security vulnerabilities.
+          <br/><br/>To save the sanitised editor input, press Acknowledge. To return to the editor without sanitising, press Cancel.`}
```

Comment:
i think adding 2 `br`s to format rather than being able to do it naturally is a sign that the abstraction boundary might be wrong here. other cases to consider might be that the warning modal has a fixed size, perhaps sometimes have picture etc; it might be worth refactoring the exposed API to give consumers more flexibility

Reply:
https://github.com/isomerpages/isomercms-frontend/pull/941/commits/bae8cb2823b64f92ac2e40cee947dd907a0baae9